### PR TITLE
Use travis native cache

### PR DESCRIPTION
**Pull request by Kojoley**
As per https://docs.travis-ci.com/user/caching/

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -1,11 +1,10 @@
...
+
 sudo: false
 
 cache:
-  directories:
-    - $HOME/.ccache
-    - $HOME/.cache/pip
-    - $HOME/.cache/matplotlib
-    - $HOME/Library/Caches/pip
+  - ccache
```
Comment (jenshnielsen) [inline .travis.yml]: It's not clear from the travis docs that `pip` caches `$HOME/Library/Caches/pip` which is the pip cache directory on OSX.

**Comment by Kojoley** [comment]
As per https://github.com/travis-ci/travis-ci/issues/4011 it is now support OSX too.

**Comment by jenshnielsen** [comment]
I know that caching is supported on OSX that's why I added `$HOME/Library/Caches/pip` to the list of cached directories. But the documentation that you link to says that doing 

```
cache:
  - pip
```
only adds

```
~/.cache/pip
```
to the cache and not `$HOME/Library/Caches/pip`

I can't see anything in the docs or the issue that you are linking to that indicates otherwise. 

If you compare the OSX build of your PR to one of the Linux ones it seems to confirm that.
Linux:
https://s3.amazonaws.com/archive.travis-ci.org/jobs/146969225/log.txt
OSX:
https://api.travis-ci.org/jobs/146969231/log.txt?deansi=true

In the osx build all the dependencies are redownloaded.

**Comment by Kojoley** [comment]
Hmm. Even ccache was not used and stored on OSX

**Comment by Kojoley** [comment]
I have found that `apt-get` and `brew` currently is not cached. Should I reuse this PR or open a new one? (this seems can be done with `cache: apt`)

**Comment by jenshnielsen** [comment]
Good catch. I think it's fine to use this one you can always change the title in the github web interface

**Comment by Kojoley** [comment]
Sigh. It is not possible now https://github.com/travis-ci/travis-ci/issues/5876

**Comment by Kojoley** [comment]
Summary:
  - There is no way to cache `apt-get` and `brew` packages with `cache: apt`
    (it is no longer exists https://github.com/travis-ci/travis-ci/issues/5876)
  - `apt-get` is fast as is ```Fetched 158 MB in 3s (46.5 MB/s)``` (sometimes it slowdowns to 15 sec)
  - Storing `brew` cache gives no or insignificant speed up (even makes cache store phase much longer)
  - `cache: ccache` is not supported with `language: python` (https://github.com/travis-ci/travis-ci/issues/5853)
  - `cache: pip` does not work on OSX
  - `global` cannot be used in `cache` (like in `env`) _(workarounded with copy-paste)_:
    ```yaml
    cache:
      global:
        pip: true
        directories:
          - $HOME/.cache/matplotlib
    
    matrix:
      include:
        - os: osx
          osx_image: xcode7.3
          cache:
            pip: false
            directories:
              - $HOME/Library/Caches/pip
    ```

I have updated `appveyor.yml` with comments about results above.

**Pull request merged**